### PR TITLE
Allow building on FreeBSD variants

### PR DESCRIPTION
**Pull request by hdholm**
These changes allow building the project on FreeBSD and it's variants.  The README is changed to note describe the build.  A number of changes are to tests to remove bash dependencies since bash is not guaranteed to be installed on FreeBSD.  Additionally while researching the openwrt changes (which are similar in many ways) I discovered that the openwrt project is using a somewhat different configuration.  And in fact tang will not run on openwrt as configured here before.  So I updated that configuration to more closely match the openwrt one.  I'm happy to make and changes or breakup the pull request if that is desired.

### CODE REVIEW

**Comment by sergio-correia** [comment]
Thanks. This looks good, in general, but I am not sure about the `BSD-MANIFEST` file. How is it going to be updated with new releases? Wouldn't it be more appropriate to have this file downstream somewhere?

**Comment by hdholm** [comment]
Perhaps.  Frankly I'm new at the build stuff; not much programming in my day job these days. I'll try to look into best practice. But I should probably make it derived from the build/install somehow - particularly the file hashes.  Thanks for the feedback.

**Comment by hdholm** [comment]
After looking into this some, it looks like the manifest is only needed to build a package outside of the ports tree.  Since my goal would be to get it incorporated into the ports tree, there should be no need for the manifest file. I can pull that file out and update my README changes if that's convenient for you.  Let me know what works best here - again I'm a little new to this kind of collaboration - should I resubmit a revised PR, or just a PR for a changeset that updates the original?  (I'll do the same for the Jose PR.)

**Comment by sergio-correia** [comment]
> After looking into this some, it looks like the manifest is only needed to build a package outside of the ports tree. Since my goal would be to get it incorporated into the ports tree, there should be no need for the manifest file. I can pull that file out and update my README changes if that's convenient for you. 

Thanks, that would be nice.

> Let me know what works best here - again I'm a little new to this kind of collaboration - should I resubmit a revised PR, or just a PR for a changeset that updates the original? (I'll do the same for the Jose PR.)

I usually just rebase my local branch and force push to my fork. The PR would then be updated to reflect these changes.

**Comment by hdholm** [comment]
I hope I've correctly addressed the changes here (and in Jose.)  Let me know if you have any issues or concerns.  It looks like the fedora:rawhide build failed in the CI actions, but that seems to be a problem with the fedora setup unrelated to the tang code itself.

**Comment by hdholm** [comment]
I know everyone is busy.  Is there some issue with the PR I need to fix, or is it just waiting on someone having time to look at it?

**Comment by sergio-correia** [comment]
> [...] or is it just waiting on someone having time to look at it?

Yeah, I will look into it again this week.

File: README.md [hunk 1]
```diff
@@ -89,7 +89,31 @@ Instead of using systemd for socket activation you can use another daemon for
...
+    (as root) service tangd start
+
+Once built it does not require the many packages above, but still requires
+jose, socat and http_parser. 
+
+FreeBSD, HardendedBSD, and OPNsense use inetd rather than systemd or
+xinetd. To limit the need to manage inetd configuration which has a shared
+config file, tangd is instead packaged to depend on `socat`.  Of course,
+if desired it may be configured to run instead from inetd.conf in which case
+the socate package will no longer be required.
```
Comment (sergio-correia) [inline README.md:116]: Typo here: `socate`

**Review by sergio-correia — COMMENTED** [review]
Okay, I finally had time to look into this again. It looks good, but there is a typo when referring to socat in the README. Please update it so that I can merge this. Thanks!

**Comment by sergio-correia** [comment]
Thanks!

**Pull request merged**